core/src/main/scala/org/ensime/core/javac/Helpers.scala

```diff
@@ -3,31 +3,21 @@
 package org.ensime.core.javac
 
 import akka.event.slf4j.SLF4JLogging
-import com.sun.source.tree.{ IdentifierTree, Tree }
+import com.sun.source.tree._
 import com.sun.source.util.TreePath
-import javax.lang.model.`type`.{ DeclaredType, PrimitiveType, TypeKind, TypeMirror }
-import javax.lang.model.element.{ ElementKind, Element, TypeElement }
-import org.ensime.api.deprecating
+import javax.lang.model.`type`._
+import javax.lang.model.element._
+import javax.lang.model.element.ElementKind._
+import scala.collection.JavaConversions._
+
 import org.ensime.core.{ DocFqn, DocSig }
+import org.ensime.indexer._
 
-@deprecating("prefer FullyQualifiedName")
-final case class JavaFqn(pack: Option[String], typename: Option[String], fieldOrMethod: Option[String]) {
-  def toDocSig = DocSig(DocFqn(pack.getOrElse(""), typename.getOrElse("")), fieldOrMethod)
-  def toFqnString = Array(pack, typename, fieldOrMethod).flatten.mkString(".")
-  def toQueryString = Array(pack, typename.map(_.replace(".", "$")), fieldOrMethod).flatten.mkString(".")
-}
+trait Helpers extends UnsafeHelpers with SLF4JLogging {
 
-object JavaFqn {
-  def apply(pack: String, tpe: String, fieldOrMethod: Option[String]): JavaFqn = {
-    JavaFqn(
-      if (pack.isEmpty) None else Some(pack),
-      if (tpe.isEmpty) None else Some(tpe),
-      fieldOrMethod
-    )
+  private implicit class EnhachedElement(e: Element) {
```

Comment:
Yeah I just hate those blown up patter matchings. Plus I'm not too good in naming things (I guess nobody is).